### PR TITLE
BC-17 # use new v2 APIs in BMP exclusively

### PR DESCRIPTION
**Pull request by jokeyrhyme**
## Added

- `bm bmp login` immediately attempts to use new credentials, and shows the result

    - does not store rejected credentials

- `bm bmp scope` also attempts to use stored credentials, and shows the result

- added instructions to use an older CLI with older BMP services when appropriate

## Changed

- BC-17: use new v2 APIs in BMP exclusively (#17, @jokeyrhyme)

- BC-17: `bm bmp login` only accepts a [JWT](https://jwt.io/), rejects anything else (#17, @jokeyrhyme)

- no more `bm bmp whoami` command, it's gone

## Review Notes

- the v1 APIs provide a way to retrieve the current user's email address, which was used for `whoami`

- the v2 APIs don't have this functionality, however, it's apparently in the JWT

    - so if we really want `whoami` back, it's possible to reimplement it by decoding the JWT

### CODE REVIEW

File: test/deploy.js [hunk 1]
```diff
@@ -52,7 +52,8 @@ test.beforeEach((t) => {
 test.serial('deployAll', (t) => {
```
Comment (simonmarklar) [inline test/deploy.js]: I think there is one more scenario to test (but maybe i cant see it here??), and I only discovered it when i was knee deep in the platform code. You can PUT  multiple interactions in one request via `PUT /answerspaces/name/interactions/1,2,3` where 1, 2,3 are the id's of the interactions to update, which must match to the objects in the `interactions` part of the payload.

Comment (jokeyrhyme) [inline test/deploy.js]: I might look into batch requests like that depending on the performance situation. This isn't as simple as just doing all interactions all at once: I'd feel more comfortable doing them in batches of 100 or something, giving it a cap. And that's what would bring the extra complexity.

Definitely worth keeping in mind though. I'd prefer to have a separate Issue / PR to try it.

**Comment by simonmarklar** [comment]
looks like i missed this, didn't see a notification...

LGTM

**Comment by simonmarklar** [comment]
looks like i missed this, didn't see a notification...

LGTM

**Pull request merged**